config/initializers/content_security_policy.rb

```diff
@@ -1,31 +1,25 @@
-# Be sure to restart your server when you modify this file.
+# frozen_string_literal: true
 
-# Define an application-wide content security policy.
-# See the Securing Rails Applications Guide for more information:
-# https://guides.rubyonrails.org/security.html#content-security-policy-header
+# Configure Content Security Policy headers
+# See: https://guides.rubyonrails.org/security.html#content-security-policy-header
 
-Rails.application.config.to_prepare do
-  Rails.application.config.content_security_policy do |policy|
-    policy.default_src '*', :data, :mediastream, :blob, :filesystem, :ws, :wss, :unsafe_eval, :unsafe_inline
+require 'three_scale/content_security_policy'
+
+Rails.application.configure do
+  if ThreeScale::ContentSecurityPolicy::AdminPortal.enabled?
```

Comment:
I think this might be a bit confusing...

So, if `admin_portal.enabled: true`, and `developer_portal.enabled: false`, the admin portal's custom policy from YAML is applied for the developer portal too.